src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -248,6 +253,32 @@ private Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibility)
             return result.Select(kvp => (kvp.Key, kvp.Value)).ToImmutableArray();
         }
 
+        /// <summary>
+        /// Get symbol names that are visible and could hide or cause a conflict with potential names.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        private ImmutableHashSet<string> GetPossibleConflictingSymbolNames(CSharpSyntaxContext context)
```

Comment:
We have helpers for generating unique non-conflicting names based on location, see http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/LanguageServices/SemanticsFactsService/ISemanticFactsService.cs,110

Reply:
Hmm - I already have names generated [from here](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/Completion/CompletionProviders/DeclarationNameCompletionProvider.NameGenerator.cs,1883522963f5d313).  These are then modified based on the styles [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs,238).

Should I replace any removed name with a generated name / change how the base name suggestions are found?

Reply:
Maybe you can extract a separate helper `ISemanticFactsService.IsUniqueName(name, ...)`? Another option is just check `GenerateUniqueName(baseName, ...) == baseName`.